src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
     LOG_INFO(_logger, "Logger initialization done");
 }
 
+bool Log::getLogEstimatedSize(uint64_t &size, IoError &ioError) {
+    const SyncPath logPath = _filePath.parent_path();
+    ioError = IoErrorSuccess;
+    for (int i = 0; i < 2; i++) {  // Retry once in case a log file is archived/created during the first iteration
+        bool tooDeep = false;
+        bool result = IoHelper::getDirectorySize(logPath, size, 0, tooDeep, ioError);
+        if (result && ioError == IoErrorSuccess) {
+            return true;
+        }
+    }
+    return false;
+}
+
+SyncPath Log::getLogFilePath() const {
+    return _filePath;
+}
+
+ExitCode Log::generateLogsSupportArchive(bool includeOldLogs, const SyncPath &outputPath, const SyncPath &archiveName,
+                                         ExitCause &exitCause, std::function<void(int)> progressCallback) {
+    // Get the log directory path
+    const SyncPath logPath = _filePath.parent_path();
+    const SyncPath tempDirectory =
+        logPath / "send_log_directory_temp" / ("tempLogArchive_" + CommonUtility::generateRandomStringAlphaNum(10));
+    exitCause = ExitCauseUnknown;
+    IoError ioError;
+
+    // Create temp folder
+    if (!IoHelper::createDirectory(tempDirectory.parent_path(), ioError) && ioError != IoErrorDirectoryExists) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::createDirectory: " << Utility::formatIoError(tempDirectory.parent_path(), ioError).c_str());
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+
+    if (!IoHelper::createDirectory(tempDirectory, ioError)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::createDirectory: " << Utility::formatIoError(tempDirectory, ioError).c_str());
+        IoError ignoreError;
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ignoreError);
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+
+    ExitCode exitCode = copyLogsTo(tempDirectory, includeOldLogs, exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Unable to copy logs to temp folder: " << exitCause);
```

Comment:
I would love to have a function named `formatExitCause` with human readable text output.

Reply:
It maybe deserve it's own PR, could be base on `QString ParametersDialog::getAppErrorText(QString fctCode, ExitCode exitCode, ExitCause exitCause)`, it already does a similar job.

Reply:
I created a ticket: https://infomaniak.atlassian.net/browse/KDESKTOP-844